### PR TITLE
Implement persistent filters

### PR DESCRIPTION
**Pull request by nrwinner**

### CODE REVIEW

**Comment by DonSeannelly** [comment]
There seems to be some odd URL formatting going on when certain filters are not selected. For instance, here's the URL when I selected one of each filter type: `/browse?text=&currPage=1&limit=20&length=course&level=k-12&&orderBy=date&sortType=-1` but when it is all cleared I have `/browse?text=&currPage=1&limit=20&&&` with a few trailing & operators.

Perhaps we could strip these off?

File: cypress/integration/browse.spec.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    beforeEach(() => {
+        // Return to home page before each test
+        cy.visit('http://localhost:4200/home');
+    });   
+    // =============================================================
+    // /browse testing
+    // =============================================================
+    it('Click a card and see details page', () => {
+        // Type in to search bar
+        cy.get('.search-input').type('Nick');
```
Comment (DonSeannelly) [inline cypress/integration/browse.spec.js]: Input values should be moved to a fixture for easy management and reuse

File: cypress/integration/browse.spec.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+ /// <reference types="cypress" />
+
+import { SSL_OP_CISCO_ANYCONNECT } from 'constants';
+
+describe('Browse', () => {
+
+    beforeEach(() => {
+        // Return to home page before each test
+        cy.visit('http://localhost:4200/home');
```
Comment (DonSeannelly) [inline cypress/integration/browse.spec.js]: Routes should also be moved to a fixture for easy updating.

File: cypress/integration/details.spec.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        // Different steps for login - don't replace with helper method
+        cy.get('input[name=username]').type('nvisal1');
+        cy.get('input[name=password]').type('122595');
+        cy.get('.auth-button').click();
+
+        // Assert URL 
+        cy.url().should('include', 'home');
+    });
+
+    // it('Navigate to details and click DOWNLOAD NOW button when logged out', () => {
```
Comment (DonSeannelly) [inline cypress/integration/details.spec.js]: Why is this test commented out?

File: src/app/auth/can-reset-password.guard.spec.ts [hunk 4]
```diff
@@ -1,15 +1,15 @@
-import { TestBed, async, inject } from '@angular/core/testing';
+// import { TestBed, async, inject } from '@angular/core/testing';
```
Comment (DonSeannelly) [inline src/app/auth/can-reset-password.guard.spec.ts]: This spec file should not be commented out.

File: src/app/core/auth.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    const loginInfo = {
+      username: 'nvisal1',
+      password: '122595'
+    };
+    return service.login(loginInfo).then(val => {
+      const status = service.status;
+      expect(status).toBe(true);
+    });
+  }));
+  // it('should return status false because logged out', inject([AuthService], (service: AuthService) => {
```
Comment (DonSeannelly) [inline src/app/core/auth.service.spec.ts]: Why is this test commented out?

File: src/app/core/auth.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    });
+  }));
+  it('should return user', inject([AuthService], (service: AuthService) => {
+    const loginInfo = {
+      username: 'nvisal1',
+      password: '122595'
+    };
+    return service.login(loginInfo).then(val => {
+      return service.checkPassword(loginInfo).then(val => {
+        console.log('hello');
```
Comment (DonSeannelly) [inline src/app/core/auth.service.spec.ts]: Is this log left over from writing the test?

File: src/app/core/auth.service.ts [hunk 7]
```diff
@@ -26,7 +26,7 @@ export class AuthService {
   constructor(
     private http: HttpClient,
     private cookies: CookieService,
-    private router: Router
+    // private router: Router
```
Comment (DonSeannelly) [inline src/app/core/auth.service.ts]: If the router is not being used anymore it and it's import should be removed.

File: src/app/core/cartv2.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  }));
+  it('should return cart', inject([CartV2Service, AuthService], (service: CartV2Service, auth: AuthService) => {
+    const loginInfo = {
+      username: 'nvisal1',
+      password: '122595'
+    };
+    return auth.login(loginInfo).then(val => {
+      service.updateUser();
+      return service.getCart().then(val => {
+        console.log(val);
```
Comment (DonSeannelly) [inline src/app/core/cartv2.service.spec.ts]: Maybe another remnant of writing the test?

File: src/app/core/user.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    return service.searchUsers({text: 'nick'}).then(val => {
+      expect(val).toBeDefined();
+    });
+  }));
+});
+
+
+
+
+// export type UserEdit = {
```
Comment (DonSeannelly) [inline src/app/core/user.service.spec.ts]: If this type/information isn't being used it should be removed. If it is informative, then maybe it should be moved into a comment block with more information.

**Comment by nvisal1** [comment]
Changes to the gateway and user service should be pushed along with this. passwordMatch function was changed from GET to POST

**Pull request merged**